packages/integrations/useProgressBar/index.md

```diff
@@ -0,0 +1,34 @@
+# useProgressBar
+
+> Reactive wrapper for [`nprogress`](https://github.com/rstacruz/nprogress)
+
+## Usage
+
+```js {6}
+import { useProgressBar } from '@vueuse/integrations'
+
+const loading = useProgressBar()
+
+function toggle() {
+  loading.value = !loading.value
+}
+```
+
+### Passing a source ref
+
+You can pass a `ref` to it, changes from of the source ref will be reflected to your loading progress.
+
+```js {4}
+import { ref } from 'vue'
+import { useProgressBar } from '@vueuse/integrations'
+
+const loading = ref(false)
+
+useProgressBar(loading)
+
+function toggle() {
+  loading.value = !loading.value
+}
+```
+
```

Comment:
Maybe we should have some statements showing that the existence the `options` arg and redirect users to check `nprogress`'s documentation.

Reply:
```js {13}
export function useNProgress(
  currentProgress: MaybeRef<number | null | undefined> = null,
  options?: NProgressOptions | undefined,
) {
  const progress = isRef(currentProgress) ? currentProgress : ref(null)
  const isLoading = ref(false)

  if (options)
    nprogress.configure(options)

  watch(isLoading, active => active ? nprogress.start() : nprogress.done())

  watch(progress, (n) => {
    if (typeof n === 'number')
      nprogress.set(n)
  })

  tryOnUnmounted(nprogress.remove)

  return {
    isLoading,
    progress,
    start: nprogress.start,
    done: nprogress.done,
    remove: nprogress.remove,
  }
}
```

What do you think?